### PR TITLE
[SYCL][Joint Matrix] Support 1x64x16 bf16 combination

### PR DESCRIPTION
**Pull request by YuriPlyakhin**
- add support in device_info
- add support in tests

### CODE REVIEW

**Review by maarquitos14 — APPROVED** [review]
LGTM.

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_packedB.cpp [hunk 1]
```diff
@@ -13,12 +13,5 @@
 
 #include "common.hpp"
 
-using namespace sycl;
-using namespace sycl::ext::oneapi::experimental::matrix;
-
 #define SG_SZ 16
```
Comment (dkhaldi) [inline sycl/test-e2e/Matrix/joint_matrix_bfloat16_packedB.cpp:16]: you should use `get_sg_size`  to get rid of this macro here

**Comment by YuriPlyakhin** [comment]
@intel/sycl-matrix-reviewers , please, review

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_packedB_impl.hpp [hunk 2]
```diff
@@ -13,7 +22,7 @@ void matrix_multiply(big_matrix<T1, M, N> &C, big_matrix<T2, M, K> &A,
      auto accA = bufA.get_access<access::mode::read_write>(cgh);
      auto accB = bufB.get_access<access::mode::read_write>(cgh);
 
-     cgh.parallel_for<class imatrix>(
+     cgh.parallel_for<kernel_name>(
          nd_range<2>({NDRangeM, NDRangeN * SG_SZ}, {1, 1 * SG_SZ}),
          [=](nd_item<2> spmd_item) [[intel::reqd_sub_group_size(SG_SZ)]]
```
Comment (dkhaldi) [inline sycl/test-e2e/Matrix/joint_matrix_bfloat16_packedB_impl.hpp:27]: Remove the attribute when  get_sg_size is used.
Keep the attribute for SG_SIZE=32

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_packedB.cpp (hunk 1 above)
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/joint_matrix_bfloat16_packedB.cpp:16]: This is not the point of this PR

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_packedB_impl.hpp (hunk 2 above)
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/joint_matrix_bfloat16_packedB_impl.hpp:27]: This is not the point of this PR

**Review by dkhaldi — APPROVED** [review]

**Comment by YuriPlyakhin** [comment]
@intel/llvm-gatekeepers , please, merge, all green.

**Pull request merged**